networkx/algorithms/tests/test_mis.py

```diff
@@ -8,78 +8,48 @@
-        """Maximal independent set: K5"""
-        G = nx.complete_graph(5)
-        for node in G:
-            assert nx.maximal_independent_set(G, [node]) == [node]
-
-    def test_K55(self):
-        """Maximal independent set: K55"""
-        G = nx.complete_graph(55)
-        for node in G:
-            assert nx.maximal_independent_set(G, [node]) == [node]
-
-    def test_exception(self):
-        """Bad input should raise exception."""
-        G = self.florentine
-        pytest.raises(nx.NetworkXUnfeasible, nx.maximal_independent_set, G, ["Smith"])
-        pytest.raises(
-            nx.NetworkXUnfeasible, nx.maximal_independent_set, G, ["Salviati", "Pazzi"]
-        )
-
-    def test_digraph_exception(self):
-        G = nx.DiGraph([(1, 2), (3, 4)])
-        pytest.raises(nx.NetworkXNotImplemented, nx.maximal_independent_set, G)
-
-    def test_florentine_family(self):
-        G = self.florentine
-        indep = nx.maximal_independent_set(G, ["Medici", "Bischeri"])
-        assert sorted(indep) == sorted(
-            ["Medici", "Bischeri", "Castellani", "Pazzi", "Ginori", "Lamberteschi"]
-        )
-
-    def test_bipartite(self):
-        G = nx.complete_bipartite_graph(12, 34)
-        indep = nx.maximal_independent_set(G, [4, 5, 9, 10])
-        assert sorted(indep) == list(range(12))
-
-    def test_random_graphs(self):
-        """Generate 50 random graphs of different types and sizes and
-        make sure that all sets are independent and maximal."""
-        for i in range(0, 50, 10):
-            G = nx.random_graphs.erdos_renyi_graph(i * 10 + 1, random.random())
-            IS = nx.maximal_independent_set(G)
-            assert not list(G.subgraph(IS).edges())
-            neighbors_of_MIS = set.union(*(set(G.neighbors(v)) for v in IS))
-            for v in set(G.nodes()).difference(IS):
-                assert v in neighbors_of_MIS
+def test_random_seed():
+    G = nx.empty_graph(5)
+    assert nx.maximal_independent_set(G, seed=1) == [1, 0, 3, 2, 4]
+
+
+@pytest.mark.parametrize("graph", [nx.complete_graph(5), nx.complete_graph(55)])
+def test_K5(graph):
+    """Maximal independent set for complete graphs"""
+    assert all(nx.maximal_independent_set(graph, [n]) == [n] for n in graph)
+
+
+def test_exceptions():
+    """Bad input should raise exception."""
+    G = nx.florentine_families_graph()
+    pytest.raises(nx.NetworkXUnfeasible, nx.maximal_independent_set, G, ["Smith"])
+    pytest.raises(
+        nx.NetworkXUnfeasible, nx.maximal_independent_set, G, ["Salviati", "Pazzi"]
+    )
+    # MaximalIndependantSet is not implemented for directed graphs
+    pytest.raises(nx.NetworkXNotImplemented, nx.maximal_independent_set, nx.DiGraph(G))
+
+
+def test_florentine_family():
+    G = nx.florentine_families_graph()
+    indep = nx.maximal_independent_set(G, ["Medici", "Bischeri"])
+    assert sorted(indep) == sorted(
+        ["Medici", "Bischeri", "Castellani", "Pazzi", "Ginori", "Lamberteschi"]
+    )
```

Comment:
If the order doesn't matter, you might be able to get away with e.g.

```suggestion
    assert set(indep) == {
        "Medici", "Bischeri", "Castellani", "Pazzi", "Ginori", "Lamberteschi"
    }
```

Of course, it's completely fine as-is now!